src/par_map.rs

```diff
@@ -0,0 +1,25 @@
+use rayon::iter::{FromParallelIterator, IntoParallelIterator, ParallelIterator};
+
+use crate::Database;
+
+pub fn par_map<Db, D, E, C>(
+    db: &Db,
+    inputs: impl IntoParallelIterator<Item = D>,
+    op: fn(&Db, D) -> E,
+) -> C
+where
+    Db: Database,
+    D: Send,
+    E: Send + Sync,
+    C: FromParallelIterator<E>,
+{
+    dbg!(db.zalsa().views());
```

Comment:
i've been doing a bunch of digging to figure out why this panics when used with `tests/parallel/parallel_map.rs` test. i initially saw that the underling `Zalza::views` was empty, so the test would naturally panic—there is nothing to downcast to! I've since added calls to `tracked_fn` in the test, and now, I'm seeing a downcast-able database.

```
[src/par_map.rs:16:5] db.zalsa().views() = DynDowncasts {
    vec: [
        DynDowncast(
            "dyn parallel_map::common::LogDatabase",
        ),
    ],
```

This makes sense! databases are discovered upon first use, but for `par_map` to work, I think the list of databases to cast to need to be reflexive: that is, they need to include themselves. I'll try making this change.